### PR TITLE
Add iati-activity/result/reference and test cases

### LINKED ISSUE #362: Add new reference element (as a child of the result element)
Add a new element iati-activities/iati-activity/result/reference with definition:

> A reference element to allow for the coded identification of a results framework.

The element will have an occurrence rule of `0..*`.

This will also add the following sub-attributes/elements with definitions:

- `iati-activities/iati-activity/result/reference/@code` - occurs `1..1`, type xsd:string
> A code from the codelist identified in vocabulary that identifies a results framework

- `iati-activities/iati-activity/result/reference/@vocabulary` - occurs `1..1`
> A code for the results framework vocabulary. The code must be a valid value in the ResultVocabulary codelist

- `iati-activities/iati-activity/result/reference/@vocabulary-uri` - occurs `0..1`, type xsd:anyURI
> The URI where this vocabulary is defined. If the vocabulary is 99 or 98 (reporting organisation), the URI where this internal vocabulary is defined. While this is an optional field it is STRONGLY RECOMMENDED that all publishers use it to ensure that the meaning of their codes are fully understood by data users.

**Tasks:**

- [x] Add test cases - Done in IATI/IATI-Schemas#383
- [x] Add `iati-activities/iati-activity/result/reference` - Done in IATI/IATI-Schemas#383
- [x] Add `iati-activities/iati-activity/result/reference/@code` - Done in IATI/IATI-Schemas#383
- [x] Add `iati-activities/iati-activity/result/reference/@vocabulary` - Done in IATI/IATI-Schemas#383
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/reference/@vocabulary` (to reference this attribute to the `ResultVocabulary` codelist)
- [x] Add `iati-activities/iati-activity/result/reference/@vocabulary-uri` - Done in IATI/IATI-Schemas#383

The addition of the new `ResultVocabulary` codelist has been [added as a new issue on the IATI-Codelists-NonEmbedded repository](https://github.com/IATI/IATI-Codelists-NonEmbedded/issues/181).

### PR DESCRIPTION
**Pull request by dalepotter**
Fixes #362

### CODE REVIEW

File: tests/should-fail/iati-activities/iati-activity/result/reference/err-not-iati-xml-missing-attribute_reference-element-present-no-required-attributes.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <activity-status code="xx"/>
+    <activity-date type="xx" iso-date="2013-11-27"/>
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx" aggregation-status="0">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <reference/><!-- FAIL: Optional `reference` element present, but both the mandatory `code` and `vocabulary` attributes are not present. -->
```
Comment (hayfield) [inline tests/should-fail/iati-activities/iati-activity/result/reference/err-not-iati-xml-missing-attribute_reference-element-present-no-required-attributes.xml]: If they are tested individually, this test file is not required.

(and if they aren't tested individually, this test file doesn't check the things that each is required)

...and there's no manner in XSD 1.0 to say that one of a pair of attributes is required https://stackoverflow.com/a/763108

File: tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-element.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <activity-date type="xx" iso-date="2013-11-27"/>
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx" aggregation-status="0">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <reference vocabulary="xx" code="xx"><!-- PASS: Optional `reference` element present, with two mandatory attributes. -->
+        <example-namespace:namespace-element></example-namespace:namespace-element><!-- Optional namespace element included. -->
```
Comment (hayfield) [inline tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-element.xml]: Could be clearer that this is what is being tested.

**Review by hayfield — DISMISSED** [review]
One redundant test file and a couple of comments that could be a tad clearer.

File: tests/should-pass/iati-activities/iati-activity/result/reference/no-reference-element-present.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/reference/no-reference-element-present.xml]: Is this needed? Isn't it covered by the minimal IATI file case?

Comment (allthatilk) [reply tests/should-pass/iati-activities/iati-activity/result/reference/no-reference-element-present.xml]: Nvm, leave for integration branch as needs to be moved to `result` level folder.

**Review by allthatilk — APPROVED** [review]

File: tests/should-fail/iati-activities/iati-activity/result/reference/err-not-iati-xml-missing-attribute_reference-element-present-no-required-attributes.xml (hunk 1 above)
Comment (dalepotter) [reply tests/should-fail/iati-activities/iati-activity/result/reference/err-not-iati-xml-missing-attribute_reference-element-present-no-required-attributes.xml]: Fixed in 1729f95.

File: tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-element.xml (hunk 2 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-element.xml]: I've made changes in 59a684c - is it clearer now?

File: tests/should-pass/iati-activities/iati-activity/result/reference/no-reference-element-present.xml (hunk 3 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/reference/no-reference-element-present.xml]: Fixed in e23a674 anyway!

**Pull request merged**